src/wrapper-utils.js

```diff
@@ -0,0 +1,191 @@
+/* global cloneInto, exportFunction, mozProxies */
+// Tests don't define this variable so fallback to behave like chrome
+const hasMozProxies = typeof mozProxies !== 'undefined' ? mozProxies : false
+const globalObj = typeof window === 'undefined' ? globalThis : window
+const getOwnPropertyDescriptor = Object.getOwnPropertyDescriptor
+const functionToString = Function.prototype.toString
+const objectKeys = Object.keys
+const NativeProxy = Proxy
+
+export function defineProperty (object, propertyName, descriptor) {
+    if (hasMozProxies) {
+        const usedObj = object.wrappedJSObject
+        const UsedObjectInterface = globalObj.wrappedJSObject.Object
+        const definedDescriptor = new UsedObjectInterface();
+        ['configurable', 'enumerable', 'value', 'writable'].forEach((propertyName) => {
+            if (propertyName in descriptor) {
+                definedDescriptor[propertyName] = cloneInto(
+                    descriptor[propertyName],
+                    definedDescriptor,
+                    { cloneFunctions: true })
+            }
+        });
+        ['get', 'set'].forEach((methodName) => {
+            if (methodName in descriptor) {
+                exportFunction(descriptor[methodName], definedDescriptor, { defineAs: methodName })
+            }
+        })
+        UsedObjectInterface.defineProperty(usedObj, propertyName, definedDescriptor)
+    } else {
+        Object.defineProperty(object, propertyName, descriptor)
+    }
+}
+
+/**
+ * add a fake toString() method to a wrapper function to resemble the original function
+ * @param {*} newFn
+ * @param {*} origFn
+ */
+function wrapToString (newFn, origFn) {
+    if (!newFn || !origFn) {
+        return
+    }
+    newFn.toString = function () {
```

Comment:
I think we had tests for this with one disabled for `toString.toString()` can you check if we can get that enabled?

Reply:
I think this will also need to be exported for mozilla code too.

Reply:
I tested it on Firefox, and it seems like it ignores custom `toString` on descriptor functions anyway, so it works either way.